creds_utils.go

```diff
@@ -0,0 +1,197 @@
+package jwt
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/nats-io/nkeys"
+)
+
+// DecorateJWT returns a decorated JWT that describes the kind of JWT
+func DecorateJWT(jwtString string) ([]byte, error) {
+	gc, err := DecodeGeneric(jwtString)
+	if err != nil {
+		return nil, err
+	}
+	return formatJwt(string(gc.Type), jwtString)
+}
+
+func formatJwt(kind string, jwtString string) ([]byte, error) {
+	templ := `-----BEGIN NATS %s JWT-----
+%s
+------END NATS %s JWT------
+`
+	w := bytes.NewBuffer(nil)
+	kind = strings.ToUpper(kind)
+	_, err := fmt.Fprintf(w, templ, kind, jwtString, kind)
+	if err != nil {
+		return nil, err
+	}
+	return w.Bytes(), nil
+}
+
+func DecorateSeed(seed []byte) ([]byte, error) {
+	w := bytes.NewBuffer(nil)
+	ts := bytes.TrimSpace(seed)
+	pre := string(ts[0:2])
+	kind := ""
+	switch pre {
+	case "SU":
+		kind = "USER"
+	case "SA":
+		kind = "ACCOUNT"
+	case "SO":
+		kind = "OPERATOR"
+	default:
+		return nil, errors.New("seed is not an operator, account or user seed")
+	}
+	header := `************************* IMPORTANT *************************
+NKEY Seed printed below can be used to sign and prove identity.
+NKEYs are sensitive and should be
+treated as secrets.
+-----BEGIN %s NKEY SEED-----
+`
+	_, err := fmt.Fprintf(w, header, kind)
+	if err != nil {
+		return nil, err
+	}
+	w.Write(ts)
+
+	footer := `
+------END %s NKEY SEED------
+*************************************************************
+`
+	_, err = fmt.Fprintf(w, footer, kind)
+	if err != nil {
+		return nil, err
+	}
+	return w.Bytes(), nil
+}
+
+var userConfigRE = regexp.MustCompile(`\s*(?:(?:[-]{3,}[^\n]*[-]{3,}\n)(.+)(?:\n\s*[-]{3,}[^\n]*[-]{3,}\n))`)
```

Comment:
Might want to collectively review and make sure it is what we all want.

Reply:
The regex works for the case of the extra `----`, but will likely fail in the case where there are missing newlines where expected, however, this prob is the same case for decorated CA certs, etc.

Reply:
I think we should take the time and make it a little more robust.

Reply:
This is the one from the Go client. Since I suck at regex, there is some website where you can try them on. I actually used a small test where I pass various inputs. I found out that this would work:
```
------
abc
------
```
but also this:
```
abc--def--- ghi ---
abc
--- ---
```
Point is that the strict requirement is it has to end with at least `---` before `\n`, but simply needs at least 3 '-' somewhere before that.
It looks also that the regex will allow empty lines before the end of the section, but not before. That is, this works:
```
--- start jwt ---
some jwt


--- end jwt --
```
will return `some jwt`, but this won't work:
```
--- start jwt ---


some jwt
--- end jwt ---

I have modeled the current behavior in the pending NATS C client PR. Whatever we chose for regex, I need to know to that the C client match that.

Reply:
This is a specific Go one. I also use rubular alot.

https://regex-golang.appspot.com/assets/html/index.html

Reply:
the java one basically says lines have to start and end with - (modulo whitespace) - but that is perhaps too permissive, definitely need to think harder on this maybe?